compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp

```diff
@@ -1126,6 +1126,14 @@ static void addLowerToLLVMGPUPasses(OpPassManager &modulePassManager,
 
   if (forROCDL) {
     // convert to ROCDL.
+    funcPassManager.addPass([&]() {
+      ConvertUnsupportedFloatArithPassOptions options;
+      // Convert arith operation with the given `source types` to `target`
+      // type.
+      options.sourceTypeStrs = {"f8E4M3FNUZ", "f8E5M2FNUZ"};
+      options.targetTypeStr = "f32";
+      return createConvertUnsupportedFloatArithPass(options);
+    });
```

Comment:
The notion of which type is "supported" depends on the combination of (target architecture, specific operation, specific operand for that operation). For example, on CDNA3, we have matrix multiplication instructions that handle f8E4M3FNUZ and f8E5M2FNUZ for the LHS and RHS operands, but not for the accumulator operand. How is that nuance reflected in the logic being introduced in this PR?

Reply:
https://github.com/llvm/llvm-project/blob/b04a980b5597c61a8df2b489c4894bc0240b8e13/mlir/lib/Dialect/Arith/Transforms/EmulateUnsupportedFloats.cpp#L122 This doesn't touch those operations for now but in future should add the pattern downstream of what it should look for based on the target. I should add the pass under a flag checking the target.

Reply:
I see... yes, it sounds like that is what is needed.

Reply:
Thanks for the info! It was constructive. I've updated the pass based on the target arch. This only populates `gfx94{*}` source and target types for now and can be updated based on the need.